ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -386,9 +389,29 @@ data PeerSelectionActions peeraddr peerconn m = PeerSelectionActions {
 
        -- | Read the current ledger state judgement
        --
-       readLedgerStateJudgement :: STM m LedgerStateJudgement
+       readLedgerStateJudgement :: STM m LedgerStateJudgement,
+
+       -- | Callback provided by consensus to inform it if the node is
+       -- connected to only local roots or also some external peers.
+       --
+       -- This is useful in order for the Bootstrap State Machine to
+       -- simply refuse to transition from TooOld to YoungEnough while
+       -- it only has local peers.
+       --
+       updateOutboundConnectionsState :: OutboundConnectionsState -> STM m ()
+
      }
 
+-- | Interfaces required by the peer selection governor, which do not need to
+-- be shared with actions and thus are not part of `PeerSelectionActions`.
+--
+-- TODO: add other `TVar`s which outbound governor is using.
+--
+data PeerSelectionInterfaces m = PeerSelectionInterfaces {
+      readUseLedgerPeers :: STM m UseLedgerPeers
+    }
```

Comment:
Nice! Perhaps move `peerSharing`, `readLocalRootPeers`, `readPeerSelectionTargets`, `readUseBootstrapPeers` and `readLedgerStateJudgement` from `PeerSelectionActions` to here since they are not actions as well?

Reply:
I noticed the following commit added more things but not the ones I mentioned here.

Reply:
They are useful in guarded decisions, so I don't think this is the right move.  Also, I didn't add `readUseLedgerPeers` and the other `StrictTVar`s to `PeerSelectionActions` because they are unnecessary in guarded decisions.